src/NUnitFramework/framework/Constraints/Comparers/EquatablesComparer.cs

```diff
@@ -15,6 +15,8 @@ internal static class EquatablesComparer
         {
             if (equalityComparer.CompareAsCollection && state.TopLevelComparison)
                 return null;
+            if (tolerance != null && tolerance.HasVariance)
```

Comment:
It looks like this is is because equatablescomparer checks don't use tolerance and can't because the interface doesn't support it. While this makes sense, I'm also curious if there's a chance we think this could break someone's test as it causes the EquatablesComparer to be skipped in this case. I've tried quite a bit to find edge cases here using built-in types and it seems fairly sound. 

One area of concern here for me though is for types that use `IEquatable` for equality comparisons between two different types, such as `string` and `Microsoft.Extensions.Primitives.StringSegment`. Such a test would require using the IEquatable comparer to pass, but the test would also currently pass if a tolerance were given, for example by appending `.Within(0)` .

```
[Test]
public void TestEquatables()
{
    var a = new Microsoft.Extensions.Primitives.StringSegment("a");
    var b = "a";
    Assert.That(a, Is.EqualTo(b).Within(0));
}
```

This change would cause such a test to fail with the message `"Tolerance is not supported for this comparison"`. I feel like this could be a good thing as it alerts about unnecessary comparison modifiers, but someone else may argue "if it doesn't apply why can't it be ignored?"

I admit the risk of this is low, but I'd be curious to hear other thoughts from the @nunit/framework-team about it. I think the risk here would be higher for custom types than for BCL types. In fact, this was the _only_ case I found in the BCL as most others would still pass using one of the other comparers in the comparer chain.

Reply:
Having slept on this, I think I understand the fix here and potential for discussion to be if we want a tolerance applied in an invalid situation to alert the user and fail the test or continue being ignored. I don't have much concern if it goes into v4, though perhaps it could be surprising or disruptive in 3.13 for someone to upgrade and find a test that once passed were to start failing. In either case, I'm curious the thoughts of the team